packages/components/table/src/table.js

```diff
@@ -0,0 +1,148 @@
+import { TSElement, unsafeCSS, html, customElementDefineHelper } from '@tradeshift/elements';
+import css from './table.css';
+import '@tradeshift/elements.status';
+import '@tradeshift/elements.document-card';
+import { VISIBILITY } from './utils/constants';
+
+customElementDefineHelper(
+	'ts-table',
+	class extends TSElement {
+		static get styles() {
+			return [TSElement.styles, unsafeCSS(css)];
+		}
+
+		static get properties() {
+			return {
+				rtl: { type: Boolean, reflect: true },
+				cols: { type: Array, reflect: true },
+				data: { type: Array, reflect: true }
+			};
+		}
+
+		constructor() {
+			super();
+			this.cols = [];
+			this.data = [];
+			this.rtl = false;
+		}
+
+		get tableHeader() {
+			const res = this.cols.map(
+				column =>
+					html`
+						<th class="${this.getHeaderSizeClass(column.size)} ${this.getVisibilityClass(column.visibility)}">
+							<div class="${this.getHeaderClasses(column)}">
+								${column.value}
+							</div>
+						</th>
+					`
+			);
+			return html`
+				${res}
+			`;
+		}
+
+		getHeaderClasses(column) {
+			switch (column.display) {
+				case 'right':
+					return 'display-right';
+				case 'left':
+					return 'display-left';
+				case 'center':
+					return 'display-center';
+				default:
+					return '';
+			}
+		}
+
+		getVisibilityClass(visibility) {
+			return visibility || VISIBILITY.BOTH;
+		}
+
+		getHeaderSizeClass(size) {
+			switch (size) {
+				case 'small':
+					return 'col-small';
+				case 'medium':
+					return 'col-medium';
+				case 'large':
+					return 'col-large';
+				default:
+					return '';
+			}
+		}
+
+		handleClick(row) {
+			this.dispatchCustomEvent('row-click', row.content);
+		}
+
+		get tableBody() {
+			/* eslint-disable lit/no-template-bind */
+			const res = this.data.map(row => {
+				return html`
+					<tr class="${row.isSelected ? 'selected' : ''}" @click="${() => this.handleClick(row)}">
+						${this.tableRow(row.content)}
+					</tr>
+				`;
+			});
+			/* eslint-enable */
+
+			return html`
+				${res}
+			`;
+		}
+
+		tableRow(content) {
+			return this.cols.map(
+				column =>
+					html`
+						<td class="${this.getVisibilityClass(column.visibility)}">
+							${this.getItem(content[column.property], column)}
+						</td>
+					`
+			);
+		}
+
+		getItem(item, column) {
```

Comment:
I suggest to rename it to something like `getTableCell` or `renderTableCell`.

Reply:
done